source/packages/services/assetlibrary/src/data/base.full.dao.ts

```diff
@@ -18,23 +18,40 @@ import { TYPES } from '../di/types';
 @injectable()
 export class BaseDaoFull {
     private _graph: structure.Graph;
+    private _conn: driver.DriverRemoteConnection | null;
 
     public constructor(
         @inject('neptuneUrl') private neptuneUrl: string,
         @inject(TYPES.GraphSourceFactory) graphSourceFactory: () => structure.Graph
     ) {
         this._graph = graphSourceFactory();
+        this._conn = null;
     }
 
-    protected getConnection(): NeptuneConnection {
+    protected async getConnection(): Promise<NeptuneConnection> {
         logger.debug(`base.full.dao getConnection: in:`);
-        const conn = new driver.DriverRemoteConnection(this.neptuneUrl, {
-            mimeType: 'application/vnd.gremlin-v2.0+json',
-            pingEnabled: false,
-        });
+
+        if (this._conn == null) {
+            logger.debug(`base.full.dao getConnection: create new connection:`);
+            this._conn = new driver.DriverRemoteConnection(this.neptuneUrl, {
+                mimeType: 'application/vnd.gremlin-v2.0+json',
+                pingEnabled: false,
```

Comment:
Since the AssetLibrary service is targeting Lambda and not a long-running instance, I'd argue that we want to maintain the existing `pingEnabled: false` behavior.

Reply:
Good point. I would usually err on the side of caution, since the timeouts are configurable and could conceivably be set to 15 minutes, but that's probably not necessary here.